File: src/waypalm-site/src/main/java/waypalm/site/web/controllers/HttpErrorsController.java
package waypalm.site.web.controllers;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.servlet.ModelAndView;

@Controller
@RequestMapping("/errors")
public class HttpErrorsController extends BaseController {
    private ModelAndView genericHttp(String code) {
        return view("errors/http." + code);
    }

    @RequestMapping({"/404", "/404.*"})
    public ModelAndView for404() {
        return genericHttp("404");
    }

    @RequestMapping({"/500", "/500.*"})
    public ModelAndView for500() {
        return genericHttp("500");
    }
}

File: src/waypalm-site/src/main/java/waypalm/site/web/controllers/HttpErrorsController.kt
package waypalm.site.web.controllers

import org.springframework.web.bind.annotation.RequestMapping
import javax.inject.Named
import javax.inject.Inject

Named
RequestMapping(array("/errors"))
public class HttpErrorsController [Inject](
        override var services: ControllerServices
): ControllerBase {
    fun genericHttp(code: String): ModelAndView {
        return view("errors/http." + code);
    }

    RequestMapping(array("/404", "/404.*"))
    public fun for404(): ModelAndView {
        return genericHttp("404");
    }

    RequestMapping(array("/500", "/500.*"))
    public fun for500(): ModelAndView {
        return genericHttp("500");
    }
}